### PR TITLE
feat: update contract and unit test coverage

### PR DESCRIPTION
**Pull request by Alex-Werner**

### CODE REVIEW

**Comment by Alex-Werner** [comment]
We will need to wait a little bit for some documentation update to have our contract update here. 
This should happen next week.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+language: node_js
+
+node_js:
+  - '10'
+  - '12'
+
+branches:
+  only:
+    - master
+    - /^v(0|[1-9]\d*)\.(0|[1-9]\d*)$/
```
Comment (shumkov) [inline .travis.yml]: We don’t need this one

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+services:
+  - docker
+
+addons:
+  apt:
+    packages:
+      - docker-ce
+
+before_install:
```
Comment (shumkov) [inline .travis.yml]: I would move it to before_deploy if we don’t have it alteady in deploy script

File: package.json [hunk 3]
```diff
@@ -24,8 +30,14 @@
   },
   "homepage": "https://github.com/dashevo/dashpay-contract#readme",
   "devDependencies": {
-    "@dashevo/dpp": "^0.9.1",
+    "@dashevo/dpp": "^0.14.0",
```
Comment (shumkov) [inline package.json]: 0.15-dev ?

File: scripts/register.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    .demandOption(['dapiAddress', 'serializedIdentity', 'identityPrivateKey']);
+
+const dashpayContractDocumentsSchema = require('../schema/dashpay.schema.json');
+
+/**
+ * Execute Dashpay contract registration
+ *
+ * @returns {Promise<void>}
+ */
+async function register() {
```
Comment (shumkov) [inline scripts/register.js]: Shouldn’t we use SDK for fetching/registration?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (Alex-Werner) [reply .travis.yml]: Borrowed from here : https://github.com/dashevo/dpns-contract/blob/v0.3-dev/.travis.yml 
Should I just get master then ?

File: .travis.yml (hunk 2 above)
Comment (Alex-Werner) [reply .travis.yml]: Done !

File: scripts/register.js (hunk 4 above)
Comment (Alex-Werner) [reply scripts/register.js]: Well, we can, but that would mean including js-dash-sdk + all it's dependencies for a slower results. 
But it is possible obviously.

File: .travis.yml (hunk 1 above)
Comment (shumkov) [reply .travis.yml]: It suppose to be three options:
- master
- regexp for dev branches (0.15-dev)
- full semver regexp for tags

File: scripts/register.js (hunk 4 above)
Comment (shumkov) [reply scripts/register.js]: I just was thinking about the common approach for everything and a single entry point for registering contracts. If we change the logic we won’t need to update all which use low-level libraries.

File: schema/dashpay.schema.json [hunk 5]
```diff
@@ -1,56 +1,188 @@
...
-        "type": "string",
-        "format": "url"
+      "encryptedPublicKey": {
+        "type": "array",
+        "byteArray": true,
+        "minItems": 96,
+        "maxItems": 96
+      },
+      "senderKeyIndex": {
+        "type": "integer"
```
Comment (shumkov) [inline schema/dashpay.schema.json]: Let's add minimum validation?

File: schema/dashpay.schema.json [hunk 6]
```diff
@@ -1,56 +1,192 @@
...
+        ]
+      }
     ],
     "properties": {
-      "about": {
-        "type": "string"
+      "toUserId": {
+        "type": "array",
+        "byteArray": true,
+        "minItems": 32,
```
Comment (shumkov) [inline schema/dashpay.schema.json:155]:
```
"contentMediaType": "application/x.dash.dpp.identifier"
```

Comment (Alex-Werner) [reply schema/dashpay.schema.json:155]: 👍

File: schema/dashpay.schema.json (hunk 5 above)
Comment (Alex-Werner) [reply schema/dashpay.schema.json]: 👍

File: .travis.yml (hunk 1 above)
Comment (Alex-Werner) [reply .travis.yml]: 👍

**Review by shumkov — APPROVED** [review]
LGTM! 👍

**Pull request merged**